index.js

```diff
@@ -7,26 +7,74 @@ var qs       = require('querystring')
 var FormData = require('form-data')
 
 function NeoCities(user, pass, opts) {
-  this.user = user
-  this.pass = pass
-  this.opts = opts || {}
+  if (typeof pass == 'object' || pass == undefined) {
+    this.key = user
+  }
+  this.opts = opts || (typeof pass == 'object' ? pass : {})
+  if (!this.opts.key && !this.key) {
+    this.user = user
+    this.pass = pass
+  }
+  else {
+    this.opts.key = this.opts.key || true
+  }
   this.url = url.parse(this.opts.url || 'https://neocities.org')
+  this.siteurl = this.opts.siteurl ? url.parse(this.ops.siteurl) : null
   this.client = this.url.protocol == 'https:' ? https : http
+  if (this.opts.key && !this.key) {
+    this.get('key', null, function (res) {
+        this.key = res.api_key
+        if (typeof this.opts.key == 'function') this.opts.key(this)
+    }, user, pass)
+  }
 }
 
-NeoCities.prototype.get = function(method, args, callback) {
-  var path = '/api/'+method
-
-  if(args)
-    path += '?'+qs.stringify(args)
+NeoCities.prototype.download = function(files, callback) {
+  var i = 0
+  var that = this
+  var returning = false
+  
+  if (!this.siteurl)
```

Comment:
A convention in using curly-braces could be considered here, including the curly-braces will make this easier to read for others following along the code.